android/modules/ui/src/java/ti/modules/titanium/ui/widget/searchbar/TiUISearchBar.java

```diff
@@ -1,180 +1,354 @@
-		params.addRule(RelativeLayout.CENTER_VERTICAL);
-		params.addRule(RelativeLayout.LEFT_OF, CANCEL_BUTTON_ID);
-		//		params.setMargins(4, 4, 4, 4);
-		layout.addView(getNativeView(), params);
-
-		params = new RelativeLayout.LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT);
-		params.addRule(RelativeLayout.ALIGN_PARENT_RIGHT);
-		params.addRule(RelativeLayout.CENTER_VERTICAL);
-		//		params.setMargins(0, 4, 4, 4);
-		layout.addView(cancelBtn, params);
-
-		setNativeView(layout);
+		});
+		searchView.setOnQueryTextFocusChangeListener(this);
+		setNativeView(searchView);
 	}
 
 	@Override
-	public void onTextChanged(CharSequence s, int start, int before, int count)
+	public void release()
 	{
-		if (this.searchChangeListener != null) {
-			this.searchChangeListener.filterBy(s.toString());
-		}
-		super.onTextChanged(s, start, before, count);
+		this.searchChangeListener = null;
+		this.closeButtonDrawable = null;
+		super.release();
 	}
 
 	@Override
-	public void processProperties(KrollDict d)
+	public void processProperties(KrollDict properties)
 	{
-		super.processProperties(d);
-		if (d.containsKey(TiC.PROPERTY_SHOW_CANCEL)) {
-			boolean showCancel = TiConvert.toBoolean(d, TiC.PROPERTY_SHOW_CANCEL, false);
-			cancelBtn.setVisibility(showCancel ? View.VISIBLE : View.GONE);
+		// Validate.
+		if (properties == null) {
+			return;
 		}
-		if (d.containsKey(TiC.PROPERTY_BAR_COLOR)) {
-			nativeView.setBackgroundColor(TiConvert.toColor(d, TiC.PROPERTY_BAR_COLOR));
+
+		// Fetch the search view.
+		SearchView searchView = getSearchView();
+		if (searchView == null) {
+			return;
 		}
-		if (d.containsKey(TiC.PROPERTY_PROMPT)) {
-			String strPrompt = TiConvert.toString(d, TiC.PROPERTY_PROMPT);
-			promptText.setText(strPrompt);
+
+		// Apply given properties to view.
+		if (properties.containsKey(TiC.PROPERTY_BAR_COLOR)) {
+			searchView.setBackgroundColor(TiConvert.toColor(properties, TiC.PROPERTY_BAR_COLOR));
+		}
+		if (properties.containsKey(TiC.PROPERTY_COLOR)) {
+			EditText editText = getEditText();
+			if (editText != null) {
+				editText.setTextColor(TiConvert.toColor(properties, TiC.PROPERTY_COLOR));
+			}
 		}
-		if (d.containsKey(TiC.PROPERTY_BACKGROUND_IMAGE)) {
-			processBackgroundImage(proxy.getProperty(TiC.PROPERTY_BACKGROUND_IMAGE), proxy);
+		if (properties.containsKey(TiC.PROPERTY_VALUE)) {
+			boolean wasIgnored = this.isIgnoringChangeEvent;
+			this.isIgnoringChangeEvent = true;
+			searchView.setQuery(TiConvert.toString(properties, TiC.PROPERTY_VALUE), false);
+			this.isIgnoringChangeEvent = wasIgnored;
+		}
+		if (properties.containsKey(TiC.PROPERTY_HINT_TEXT)) {
+			searchView.setQueryHint(properties.getString(TiC.PROPERTY_HINT_TEXT));
+		}
+		if (properties.containsKey(TiC.PROPERTY_HINT_TEXT_COLOR)) {
+			EditText editText = getEditText();
+			if (editText != null) {
+				editText.setHintTextColor(TiConvert.toColor(properties, TiC.PROPERTY_HINT_TEXT_COLOR));
```

Comment:
Should setting this to `undefined` return it to its system default color?

Reply:
Hmm... we don't support this for most views/properties. I don't know if it's worth the effort.

That said, in Titanium 10.0.0, I did add support for Android attribute color string references `"?attr/*"` so that you can leverage the current theme's color. I thought this might be useful for devs that use our semantic colors feature. That way you can assign the view's default theme colors like the below.
```javascript
Ti.UI.createSearchBar({
	barColor: OS_IOS ? "white" : "?android:attr/colorBackground",
	color: OS_IOS ? "black" : "?android:attr/textColor",
	hintTextColor: OS_IOS ? "gray" : "?android:attr/textColorHint",
});
```